src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/CNNVariantWriteTensors.java

```diff
@@ -0,0 +1,160 @@
+ *      <li>The bam-file argument is necessary to write 2D tensors which incorporate read data.</li>
+ * </ul>
+ *
+ * <h3>Outputs</h3>
+ * <ul>
+ * <li>data-dir This directory is created and populated with variant tensors.
+ *  it will be divided into training, validation and test sets and each set will be further divided into
+ *  positive and negative SNPs and INDELs.</li>
+ * </ul>
+ *
+ * <h3>Usage example</h3>
+ *
+ * <h4>Write Reference Tensors</h4>
+ * <pre>
+ * gatk CNNVariantWriteTensors \
+ *   -R reference.fasta \
+ *   -V input.vcf.gz \
+ *   -truth-vcf platinum-genomes.vcf \
+ *   -truth-bed platinum-confident-region.bed \
+ *   -tensor-name reference \
+ *   -data-dir my-tensor-folder
+ * </pre>
+ *
+ * <h4>Write Read Tensors</h4>
+ * <pre>
+ * gatk CNNVariantWriteTensors \
+ *   -R reference.fasta \
+ *   -V input.vcf.gz \
+ *   -truth-vcf platinum-genomes.vcf \
+ *   -truth-bed platinum-confident-region.bed \
+ *   -tensor-name read_tensor \
+ *   -bam-file input.bam \
+ *   -data-dir my-tensor-folder
+ * </pre>
+ *
+ */
+@CommandLineProgramProperties(
+        summary = "Write variant tensors for training a CNN to filter variants",
+        oneLineSummary = "Write variant tensors for training a CNN to filter variants",
+        programGroup = VariantEvaluationProgramGroup.class
+)
+@DocumentedFeature
+@ExperimentalFeature
+public class CNNVariantWriteTensors extends CommandLineProgram {
+
+    @Argument(fullName = StandardArgumentDefinitions.REFERENCE_LONG_NAME,
+            shortName = StandardArgumentDefinitions.REFERENCE_SHORT_NAME,
+            doc = "Reference fasta file.")
+    private String reference = "";
+
+    @Argument(fullName = StandardArgumentDefinitions.VARIANT_LONG_NAME,
+            shortName = StandardArgumentDefinitions.VARIANT_SHORT_NAME,
+            doc = "Input VCF file")
+    private String inputVcf = "";
+
+    @Argument(fullName = "data-dir", shortName = "data-dir", doc = "Directory of training tensors. Subdivided into train, valid and test sets.")
+    private String dataDir = "";
+
+    @Argument(fullName = "truth-vcf", shortName = "truth-vcf", doc = "Validated VCF file.")
+    private String truthVcf = "";
+
+    @Argument(fullName = "truth-bed", shortName = "truth-bed", doc = "Confident region of the validated VCF file.")
+    private String truthBed = "";
+
+    @Argument(fullName = "bam-file", shortName = "bam-file", doc = "BAM or BAMout file to use for read data when generating 2D tensors.", optional = true)
+    private String bamFile = "";
+
+    @Argument(fullName = "tensor-name", shortName = "tensor-name", doc = "Name of the tensors to generate.")
+    private TensorMapEnum tensorMap = TensorMapEnum.reference;
+
+    @Argument(fullName = "channels-last", shortName = "channels-last", doc = "Store the channels in the last axis of tensors, tensorflow->true, theano->false", optional = true)
+    private boolean channelsLast = true;
```

Comment:
If this is Theano/TF issue, it seems like it shouldn't be under user control ? Also, except in some specific cases, we usually try to define the polarity of booleans so they can have a default value of false, and can then be turned on as a flag.

Reply:
Making this advanced for now... I'd like to take it totally out of the user's hands but we still use both backends sometimes and it's helpful to be able to set it here.